common/OpTestIPMI.py

```diff
@@ -308,6 +308,29 @@ def ipl_wait_for_working_state(self, timeout=10):
         return BMC_CONST.FW_SUCCESS
 
 
+    def ipmi_ipl_wait_for_login(self, l_con, timeout=10):
+        l_rc = l_con.expect_exact(BMC_CONST.IPMI_SOL_CONSOLE_ACTIVATE_OUTPUT, timeout=120)
+        if l_rc == 0:
+            print "IPMI: sol console activated"
+        else:
+            l_msg = "Error: not able to get IPMI console"
+            raise OpTestError(l_msg)
+        time.sleep(BMC_CONST.SHORT_WAIT_IPL)
+        l_con.send("\r")
+        time.sleep(BMC_CONST.SHORT_WAIT_IPL)
+        l_rc = l_con.expect_exact(BMC_CONST.IPMI_CONSOLE_EXPECT_ENTER_OUTPUT, timeout=120)
+        if l_rc == BMC_CONST.IPMI_CONSOLE_EXPECT_LOGIN:
+            return BMC_CONST.FW_SUCCESS
+        elif l_rc in BMC_CONST.IPMI_CONSOLE_EXPECT_PETITBOOT:
+            return BMC_CONST.FW_SUCCESS
+        elif l_rc in BMC_CONST.IPMI_CONSOLE_EXPECT_RANDOM_STATE:
+            l_msg = "Error: system is in random state"
+            raise OpTestError(l_msg)
+        else:
+            l_con.expect(pexpect.TIMEOUT, timeout=30)
```

Comment:
I'm a little confused on this logic.  The general else clause is to now expect a TIMEOUT, print that out, then just continue as if all is good?  Is this the workaround for the FSP system?